### DISCUSSION TITLE
HOWTO: Migrate the media from S3ish to S3ish.

### DISCUSSION THREAD

**Discussion by shleeable** [mastodon/mastodon#27962] [Q&A]
There are a lot of people considering moving from S3 provider to S3 provider....

We need a best practice to setup an CNAME on DNS at a minimum to allow people to migrate easily.. 

but for situations were I'm using
```
S3_ENABLED=true
S3_PROTOCOL=https
S3_BUCKET=files.mastodonish.social
S3_REGION=ap-southeast-2
S3_HOSTNAME=s3-ap-southeast-2.amazonaws.com
```

How do I move to another S3 provider without breaking the previous posts media references.

https://s3-ap-southeast-2.amazonaws.com/files.aus.social/media_attachments/files/000/853/391/original/dcfd8c4e7f22ad25.jpeg

**Comment by elaine-jackson** [comment]
I think the solution here is to add an option to the .env.production file which gives server owners the option to publish a URL that redirects to the current bucket. For example when you create a toot and send it something like `mediaUrl=https://mastodon.social/filename` which returns a 302 to the current bucket. This will increase the time to first byte having to hit a redirect before the actual file, it'll also send more requests to the origin server and eat at its available bandwidth pool. Alternatively we can take the current approach and just link directly but it'll lead to broken media overtime, that's the nature of the internet though. Unless you're changing providers everyday it's not really that big of a deal though.

**Comment by flaki** [comment]
I think the [docs' recommendation of proxying S3 storage through one's owned domain](https://docs.joinmastodon.org/admin/optional/object-storage-proxy/) already resolved this?
